test/libyul/SyntaxTest.cpp

```diff
@@ -0,0 +1,145 @@
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "libyul/AsmAnalysis.h"
+#include "libyul/AsmAnalysisInfo.h"
+#include <libyul/AsmParser.h>
+
+#include "libyul/backends/wasm/WasmDialect.h"
+#include <libyul/backends/evm/EVMDialect.h>
+#include <libyul/backends/wasm/WasmDialect.h>
+
+#include "liblangutil/EVMVersion.h"
+#include <liblangutil/Exceptions.h>
+
+#include <test/libyul/SyntaxTest.h>
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::util;
+using namespace solidity::langutil;
+using namespace solidity::yul::test;
+
+namespace
+{
+std::map<string const, yul::Dialect const& (*)(langutil::EVMVersion)> validDialects = {
+	{
+		"evm",
+		[](langutil::EVMVersion _evmVersion) -> yul::Dialect const&
+		{ return yul::EVMDialect::strictAssemblyForEVM(_evmVersion); }
+	},
+	{
+		"evmTyped",
+		[](langutil::EVMVersion _evmVersion) -> yul::Dialect const&
+		{ return yul::EVMDialectTyped::strictAssemblyForEVM(_evmVersion); }
+	},
+	{
+		"yul",
+		[](langutil::EVMVersion) -> yul::Dialect const&
+		{ return yul::Dialect::yulDeprecated(); }
+	},
+	{
+		"ewasm",
+		[](langutil::EVMVersion) -> yul::Dialect const&
+		{ return yul::WasmDialect::instance(); }
+	}
+};
+
+vector<string> const validDialectNames()
+{
+	vector<string> names{size(validDialects), ""};
+	transform(begin(validDialects), end(validDialects), names.begin(), [](auto const& dialect) { return dialect.first; });
+
+	return names;
+}
+}
+
+void SyntaxTest::parseAndAnalyze()
+{
+	string dialectName = m_validatedSettings.count("Dialect") ? m_validatedSettings["Dialect"] : "evmTyped";
+
+	yul::Dialect const& dialect = validDialects[dialectName](m_evmVersion);
+
+	for (auto const& pair: m_sources)
+		try
+		{
+			ErrorList errorList{};
+			ErrorReporter errorReporter{errorList};
+
+			auto scanner = make_shared<Scanner>(CharStream(pair.second, pair.first));
+			auto parserResult = yul::Parser(errorReporter, dialect).parse(scanner, false);
+
+			if (parserResult)
+			{
+				yul::AsmAnalysisInfo analysisInfo;
+				yul::AsmAnalyzer(analysisInfo, errorReporter, dialect).analyze(*parserResult);
+			}
+
+			for (auto const& error: errorList)
+			{
+				int locationStart = -1, locationEnd = -1;
+
+				if (auto location = boost::get_error_info<errinfo_sourceLocation>(*error))
+				{
+					locationStart = location->start;
+					locationEnd = location->end;
+				}
+
+				m_errorList.emplace_back(SyntaxTestError{
+					error->typeName(),
+					errorMessage(*error),
+					pair.first,
+					locationStart,
+					locationEnd
+				});
+			}
+		}
+		catch (UnimplementedFeatureError const& _e)
```

Comment:
Do we depend on unimplemented features? Shouldn't it always be a hard failure?

Reply:
We *do* test for unimplemented feature errors in some of the syntax tests and I think it makes a lot of sense. This was intentionally added.
This became necessary after we started running codegen for all parsing/syntax tests (think fix-point types) and I think it's a good idea in general (it makes sense to be able to write tests up front and especially to test syntax leading to unimplemented feature errors), so I'd say we should keep supporting this as isoltest expectation.

Reply:
Ah sorry, this is the yul-specific part - I mistakenly thought this was common between yul and solidity.

For yul we actually really *don't* want and need this :-D.